packages/autorest.python/test/unittests/test_serialization.py

```diff
@@ -0,0 +1,2590 @@
+
+import storage_models
+
+import pytest
+
+class Resource(Model):
+    """Resource
+
+    :param str id: Resource Id
+    :param str name: Resource name
+    :param str type: Resource type
+    :param str location: Resource location
+    :param dict tags: Resource tags
+    """
+
+    _validation = {
+        'location': {'required': True},
+    }
+
+    _attribute_map = {
+        'id': {'key': 'id', 'type': 'str'},
+        'name': {'key': 'name', 'type': 'str'},
+        'type': {'key': 'type', 'type': 'str'},
+        'location': {'key': 'location', 'type': 'str'},
+        'tags': {'key': 'tags', 'type': '{str}'},
+    }
+
+    def __init__(self, location, id=None, name=None, type=None, tags=None, **kwargs):
+        super(Resource, self).__init__(**kwargs)
+        self.id = id
+        self.name = name
+        self.type = type
+        self.location = location
+        self.tags = tags
+
+
+class GenericResource(Resource):
+    """
+    Resource information.
+
+    :param str id: Resource Id
+    :param str name: Resource name
+    :param str type: Resource type
+    :param str location: Resource location
+    :param dict tags: Resource tags
+    :param Plan plan: Gets or sets the plan of the resource.
+    :param object properties: Gets or sets the resource properties.
+    """
+
+    _validation = {}
+
+    _attribute_map = {
+        'id': {'key': 'id', 'type': 'str'},
+        'name': {'key': 'name', 'type': 'str'},
+        'type': {'key': 'type', 'type': 'str'},
+        'location': {'key': 'location', 'type': 'str'},
+        'tags': {'key': 'tags', 'type': '{str}'},
+        'plan': {'key': 'plan', 'type': 'Plan'},
+        'properties': {'key': 'properties', 'type': 'object'},
+    }
+
+    def __init__(self, location, id=None, name=None, type=None, tags=None, plan=None, properties=None):
+        super(GenericResource, self).__init__(location, id=id, name=name, type=type, tags=tags)
+        self.plan = plan
+        self.properties = properties
+
+class TestModelDeserialization(unittest.TestCase):
+
+    def setUp(self):
+        self.d = Deserializer({'Resource':Resource, 'GenericResource':GenericResource})
+        return super(TestModelDeserialization, self).setUp()
+
+    def test_model_kwargs(self):
+
+        class MyModel(Model):
+            _validation = {
+                'id': {'readonly': True},
+                'name': {'required': True},
+            }
+            _attribute_map = {
+                'id': {'key': 'id', 'type': 'str'},
+                'name': {'key': 'name', 'type': 'str'},
+                'location': {'key': 'location', 'type': 'str'},
+            }
+            def __init__(self, **kwargs):
+                super(MyModel, self).__init__(**kwargs)
+                self.id = None
+                self.name = kwargs.get('name', None)
+                self.location = kwargs.get('location', None)
+
+        # validation = MyModel().validate()
```

Comment:
I am not sure whether we need to add `.validate` to keep same as msrest: https://github.com/Azure/msrest-for-python/blob/af41991f17445e4cb26d39c247f091ddfd7027b9/msrest/serialization.py#L231-L238

Reply:
`.validate()` has been removed when we vendored the file (for better or worse, we got some customers complaints). 

I would mark those tests as `pytest.skip`, as client side validation is not our priority.

Reply:
got it